src/features/Signup/ui/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from "react"
+import { Alert } from "react-bootstrap"
+import { useNavigate } from "react-router-dom"
+
+import { AuthSchema, SuccessSignupResponse, useAuth, UserStoreSchema, useSignupMutation } from "~/entities"
+import { BasicFormProvider, isObjectEmpty, Input, Checkbox, BasicButton, useCustomForm, ROUTES } from "~/shared"
+
+import { TERMS_URL } from "../lib/constants"
+import { useSignupTranslation } from "../lib/useSignupTranslation"
+import { SignupFormSchema, TSignupForm } from "../model/signup.schema"
+
+export const SignupForm = () => {
+  const navigate = useNavigate()
+  const { t } = useSignupTranslation()
+
+  const { setUserAndAuth } = useAuth()
+  const [terms, setTerms] = useState<boolean>(false)
+
+  const form = useCustomForm(
+    { defaultValues: { email: "", firstName: "", lastName: "", password: "", confirmPassword: "" } },
+    SignupFormSchema,
+  )
+  const {
+    handleSubmit,
+    formState: { errors },
+    setError,
+  } = form
+
+  const onSuccess = ({ data }: SuccessSignupResponse) => {
+    const { account, authToken, ...rest } = data
+
+    const parsedUser = UserStoreSchema.parse(rest)
+    const parsedAuth = AuthSchema.parse(authToken)
+    setUserAndAuth(parsedUser, parsedAuth)
+    navigate(ROUTES.dashboard.path)
+  }
+
+  const {
+    mutate: signup,
+    isError,
+    error,
+  } = useSignupMutation({
+    onSuccess,
+  })
+
+  const onSignupSubmit = (data: TSignupForm) => {
+    if (data.confirmPassword !== data.password) {
+      return setError("confirmPassword", { type: "pattern", message: t("passwordsUnmatched") || "" })
```

Comment:
Please use zod's `refine` function in scheme instead 
https://github.com/colinhacks/zod#refine

Reply:
Done